src/test/java/com/cdancy/bitbucket/rest/TestUtilities.java

```diff
@@ -161,14 +163,47 @@ public static synchronized GeneratedTestContents initGeneratedTestContents(final
             assertThat(project).isNotNull();
             assertThat(project.errors().isEmpty()).isTrue();
         }
+        return new GeneratedTestProject(project, projectPreviouslyExists);
+    }
 
+    /**
+     * Create a repository.
+     *
+     * @param project A Bitbucket project to host the test repository to be created.
+     * @return Repository to use for testing.
+     */
+    public static synchronized Repository createTestRepository(final BitbucketApi api, final Project project) {
         // create test repo
         final String repoKey = randomStringLettersOnly();
         final CreateRepository createRepository = CreateRepository.create(repoKey, true);
-        final Repository repository = api.repositoryApi().create(projectKey, createRepository);
+        final Repository repository = api.repositoryApi().create(project.key(), createRepository);
         assertThat(repository).isNotNull();
         assertThat(repository.errors().isEmpty()).isTrue();
+        return repository;
+    }
+
+    /**
+     * Initialize live test contents.
+     *
+     * @param endpoint Bitbucket endpoint.
+     * @param credential Bitbucket credential string.
+     * @param api Bitbucket api object.
+     * @return GeneratedTestContents to use.
+     */
+    public static synchronized GeneratedTestContents initGeneratedTestContents(final String endpoint,
+                                                                                final BitbucketAuthentication credential,
+                                                                                final BitbucketApi api) {
+        assertThat(endpoint).isNotNull();
+        assertThat(credential).isNotNull();
+        assertThat(api).isNotNull();
+
+        // Use the user provided test project, or create one
+        GeneratedTestProject generatedTestProject = maybeCreateTestProject(endpoint, credential, api);
+
+        // create test repo
+        Repository testRepository = createTestRepository(api, generatedTestProject.project);
```

Comment:
@cdancy the code would let the test call `createRepository` without adding content, thus it would be suitable for testing the empty branch scenario.